WordPress/gutenberg
# Position Panel: Open by default if a position type is set

## What?


Part of https://github.com/WordPress/gutenberg/issues/47043

Set the Position Panel in the block inspector to open by default if a value is set for the position type.

## Why?


As raised in reviews of https://github.com/WordPress/gutenberg/pull/49085, it'd be good to improve the visibility of "sticky" in the UI when the value is set. One approach for the list view is being explored over in #49122, however I thought it'd be good to improve the visibility within the sidebar inspector, too. While usually we like to deprioritise Position by keeping the panel closed by default, if a value is set, I think it'd make sense to have the panel exposed by default — a little bit like how ToolsPanelItems are displayed if a value is set.

It's a fairly subtle change, but I think will improve visibility as users are clicking around between blocks in the editor.

## How?


* Pass down `clientId` to the settings tab in the block inspector (it's already being passed down to the styles tab)
* Then, pass the `clientId` down to the `PositionControls` component (the position controls panel)
* Get the block's attributes, and check whether or not the position type is set. If so, set the panel to open by default.

## Testing Instructions





1. With TT3 theme applied, add a couple of Group blocks to a post, each with another block inside to make it easier to test.
2. Set the top-most block to Sticky
3. Click between the two Group blocks — the bottom one should have the Position panel closed by default, and the top one should have the Position panel open by default
4. Set the top-most block to Default position
5. Click to the other Group block and back to the top-most block — the Position panel should be closed by default again

## Screenshots or screencast 

The following screengrab demonstrates clicking between a Group block with no position set, and a Group block with Sticky position set. Observe the state of the Position panel at the right-hand side, and its default state when the Group block is selected:

https://user-images.githubusercontent.com/14988353/225811323-c03d8926-21a3-457c-8401-e50adba90d6c.mp4
